### ISSUE TITLE
OS X Soundcloud API Search Stopped Working

### ISSUE THREAD

**Issue by niftylettuce** [clementine-player/Clementine#4794]
Doesn't work it seems

**Comment by Chocobozzz** [comment]
Please read the [README](https://github.com/clementine-player/Clementine/blob/master/README.md#report-a-bug) and modify your issue with useful informations.

**Comment by ArnaudBienner** [comment]
I can reproduce this here too, on OS X:
13:09:05.182 ERROR SoundCloudService:391            Error when retrieving SoundCloud results: "SSL handshake failed" " (6)" 
No problem on Windows, haven't tried Linux yet.

**Comment by Chocobozzz** [comment]
I don't have any problems with the latest build on ArchLinux.

**Comment by ArnaudBienner** [comment]
The problem reoccured for me, but this time I have no idea what is wrong: the full certificates chain is still added to Qt at startup so nothing else should be missing.
If someone has an idea ( @hatstand or @davidsansome ?)

**Comment by ArnaudBienner** [comment]
Closing this issue since I believe it won't impact a lot of people: I tested on two other Mac and it worked fine.
For the record, here are the result of my investigation:
Previous commit 99521ac15f58768d29d57164ce8cca9b313c957b actually fixed the issue by adding the missing certificates, getting rid of the "ssl handshake" error when connection to soundcloud.
BUT it seems Qt4 has troubles when too many certificates are installed: seems like some will be ignored then (so the "ssl handshake" error came back again for me).
With a small reproducer:
* Using setDefaultCaCertificates instead of addDefaultCaCertificates makes the soundcloud requests work.
* Getting the defaultCaCertificates list, removing half of them (no matter which ones are removed), adding soundcloud certificates to the (smaller) list, and setting the defaultCaCertificates to this new list makes the soundcloud requests work.
* Same code with Qt5 works without any problem, no matter how many certificates are installed.

So, in conclusion, commit 99521ac15f58768d29d57164ce8cca9b313c957b is OK, problem will be fixed in Qt5 and for now I expect only OS X people with too many certificates installed to be impacted by this problem.
An alternative would be to change soundcloud service to set the defaultCertificates on the QNetworkrequest/reply objects we use there, but I would prefer to avoid changing this (now stable) code since the problem don't affect lot of people (for now I was the only person to complain about this)
